src/Compilers/CSharp/Portable/Errors/ErrorCode.cs

```diff
@@ -1232,17 +1232,20 @@ internal enum ErrorCode
         ERR_FixedBufferTooManyDimensions = 7092,
         ERR_CantReadConfigFile = 7093,
         ERR_BadAwaitInCatchFilter = 7094,
-        WRN_FilterIsConstant = 7095,
+        // WRN_FilterIsConstant = 7095,   // replaced by FilterIsConstantFalse and FilterIsConstantTrue
         ERR_EncNoPIAReference = 7096,
         //ERR_EncNoDynamicOperation = 7097,   // dynamic operations are now allowed
         ERR_LinkedNetmoduleMetadataMustProvideFullPEImage = 7098,
         ERR_MetadataReferencesNotSupported = 7099,
         ERR_InvalidAssemblyCulture = 7100,
         ERR_EncReferenceToAddedMember = 7101,
         ERR_MutuallyExclusiveOptions = 7102,
+        WRN_FilterIsConstantFalse = 7103,
+        WRN_FilterIsConstantTrue = 7104,
+        WRN_FilterIsConstantRedundantTryCatch = 7105,
```

Comment:
💭 I question the value of this specifically.

1. Between `FilterIsConstantFalse` and `FilterIsConstantRedundantTryCatch`, I don't foresee developers getting confused by the former but suddenly knowing how to proceed with the latter.
2. Distinguishing the cases introduces additional complexity in the compiler.
3. Distinguishing the cases introduces maintainability overhead since upgrade scenarios consider cases where warnings are treated as errors.

It boils down to paying a cost (complexity, maintainability) for a feature that no one will actually need in the end.

:bulb: If you keep this one, it should probably be renamed to `WRN_FilterIsConstantFalseRedundantTryCatch` (added `False`).